### PR TITLE
Python3Packages.versiontag: init at 1.2.0

### PR DESCRIPTION
**Pull request by MaskedBelgian**
###### Motivation for this change
library design to make accessing the current version number of your software easy.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6688,6 +6688,12 @@
     githubId = 775189;
     name = "Jordi Masip";
   };
+  MaskedBelgian = {
```
Comment (fabaff) [inline maintainers/maintainer-list.nix:6691]: Please make this addition in a separate commit that is part of this PR. Thanks

File: pkgs/development/python-modules/versiontag/default.nix [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+buildPythonPackage rec {
+  pname = "versiontag";
+  version = "1.2.0";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1q6mnmqdfapzd4vfr885a6i7v12ssq1akyd3d11wy2mgp9v24wzh";
+  };
+
+  meta = with lib; {
+    description = "Ultra-simple library design to make accessing the current version number of your software easy";
```
Comment (fabaff) [inline pkgs/development/python-modules/versiontag/default.nix:12]: Opinionated change

```suggestion
    description = "Python library designed to make accessing the current version number of your software";
```

**Review by fabaff — COMMENTED** [review]
- The upstream repo contains [tests](https://github.com/thelabnyc/python-versiontag/tree/master/versiontag/tests). Use `fetchFromGitHub` to get the source
- Add `pythonImportsCheck` as additional safe-guard even if the tests are executed
-  Add `versiontag` to `python-packages.nix`

File: pkgs/development/python-modules/versiontag/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "thelabnyc";
+    repo = "python-versiontag";
+    rev = "r1.2.0";
+    sha256 = "1axv2214ykgv5adajv10v2zy5fr9v77db54rkik6ja29p66zl90n";
+  };
+
+  checkInputs = [ git ];
+  buildInputs = [ setuptools];
```
Comment (risicle) [inline pkgs/development/python-modules/versiontag/default.nix:15]: Please fix the formatting here or it will forever irk the obsessives.

File: pkgs/development/python-modules/versiontag/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ lib, buildPythonPackage, fetchFromGitHub, setuptools, git }:
+
+buildPythonPackage rec {
+  pname = "versiontag";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "thelabnyc";
+    repo = "python-versiontag";
+    rev = "r1.2.0";
```
Comment (risicle) [inline pkgs/development/python-modules/versiontag/default.nix:10]: I'd suggest using `"r${version}"` for easier bumping.

**Comment by risicle** [comment]
Builds happily, macos 10.15 & nixos x86_64

**Comment by fabaff** [comment]
> Dunno if you want me to squash some commits ?

Yes please but keep the commit for `maintainer-list.nix `.

**Comment by MaskedBelgian** [comment]
> > Dunno if you want me to squash some commits ?
> 
> Yes please but keep the commit for `maintainer-list.nix `.

It's done. Thanks for the help !

**Comment by MaskedBelgian** [comment]
Hello,

Is there something else I should modify/test ?

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 135668` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.versiontag</li>
    <li>python39Packages.versiontag</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/135668#issuecomment-911434791)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-135668/results/python38Packages.versiontag/lib/python3.8/site-packages]$ python3
Python 3.8.11 (default, Jun 28 2021, 10:57:31) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import versiontag
>>> print(versiontag.get_version())
versiontag could not determine package version using cwd /nix/store/5lqzgj40fva7hzb6m5ci7yjbxd8vpcqw-python3.8-versiontag-1.2.0/lib/python3.8/site-packages. Returning default: r0.0.0
r0.0.0
```

Looks good to me.

File: pkgs/development/python-modules/versiontag/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "thelabnyc";
+    repo = "python-versiontag";
+    rev = "r${version}";
+    sha256 = "1axv2214ykgv5adajv10v2zy5fr9v77db54rkik6ja29p66zl90n";
+  };
+
+  checkInputs = [ git ];
+  buildInputs = [ setuptools ];
```
Comment (fabaff) [inline pkgs/development/python-modules/versiontag/default.nix:15]: Ups, missed that one. `setuptools` are usually not needed to be part of `buildInputs`.

Comment (MaskedBelgian) [reply pkgs/development/python-modules/versiontag/default.nix]: You're right, it builds and run without setuptools.
Changes have been done.

**Comment by MaskedBelgian** [comment]
Result of `nixpkgs-review pr 135668` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.versiontag</li>
    <li>python39Packages.versiontag</li>
  </ul>
</details>

File: pkgs/development/python-modules/versiontag/default.nix (hunk 5 above)
Comment (MaskedBelgian) [reply pkgs/development/python-modules/versiontag/default.nix]: You're right, it builds and runs without it.
Changes have been made.

**Review by jonringer — COMMENTED** [review]
I'm a little concerned about upstream, they haven't had any activity on their github for ~4 years, https://github.com/thelabnyc/python-versiontag

This usually isn't an issue, but python interpreter deprecations still need to be taken care of.
